### PR TITLE
Add file input adapter(BinaryInput)

### PR DESCRIPTION
**Pull request by bojiang**
## Description

Binary file input adapter

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation
- [ ] Test, CI, or build
- [ ] None

## Components (if applicable)
- [ ] BentoService (model packaging, dependency management, handler definition)
- [ ] Model Artifact (model serialization, multi-framework support)
- [x] Model Server (mico-batching, logging, metrics, tracing, benchmark, OpenAPI)
- [ ] YataiService (model management, deployment automation)
- [ ] Documentation

## Checklist:

- [x] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] My change requires a change in [bentoml/gallery](https://github.com/bentoml/gallery) example notebooks
- [ ] I have sent a pull request to [bentoml/gallery](https://github.com/bentoml/gallery) to make that change

### CODE REVIEW

**Comment by Sharathmk99** [comment]
Can i help here to complete the file adapter implementation?

**Comment by parano** [comment]
@bojiang is there anything blocking this PR?

**Comment by bojiang** [comment]
@parano Ready for review now.

Features that may be controversial:
- [ ] The default output in the case of CLI is `str` rather than `json`.
- [ ] The parameter accepted by the user service is a list of stream object, which can save memory when opening large files and a large number of files.

File: bentoml/adapters/file_input.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+
+class FileInput(BaseInputAdapter):
+    """Transform incoming file data from http request, cli or lambda event into numpy
+    array.
+
+    Handle incoming file data from different sources, transform them into numpy array
+    and pass down to user defined API functions
+
+    Args:
```
Comment (parano) [inline bentoml/adapters/file_input.py:40]: @bojiang could you add some more usage examples here? including a sample BentoService class using this adapter and sample curl request and CLI command invoking the API using this input adapter, similar to the one we had in multi_image_adapter here: https://github.com/bentoml/BentoML/blob/master/bentoml/adapters/multi_image_input.py#L54

**Comment by parano** [comment]
@bojiang will merge the PR for forks to try it out on master, feel free to add the inline documentation mentioned above in follow up PR.

**Pull request merged**